lib/sisimai/bite/email/zoho.rb

```diff
@@ -0,0 +1,167 @@
+      Re0 = {
+        :'from'     => %r/mailer-daemon[@]mail[.]zoho[.]com\z/,
+        :'subject'  => %r{\A(?:
+             Undelivered[ ]Mail[ ]Returned[ ]to[ ]Sender
+            |Mail[ ]Delivery[ ]Status[ ]Notification
+            )
+        }x,
+        :'x-mailer' => %r/\AZoho Mail\z/,
+      }
+      Re1 = {
+        :begin  => %r/\AThis message was created automatically by mail delivery/,
+        :rfc822 => %r/\AReceived:[ \t]*from mail[.]zoho[.]com/,
+        :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ReFailure = {
+        expired: %r/Host not reachable/
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Zoho Mail: https://www.zoho.com'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+
+      # X-ZohoMail: Si CHF_MF_NL SS_10 UW48 UB48 FMWL UW48 UB48 SGR3_1_09124_42
+      # X-Zoho-Virus-Status: 2
+      # X-Mailer: Zoho Mail
+      def headerlist;  return ['X-ZohoMail']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Zoho Mail
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['x-zohomail']
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        qprintable = false
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:'message-rfc822']).zero?` instead of `readcursor & Indicators[:'message-rfc822'] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->